File: src/services/inspectionItem.ts
import request from '../api/request'
import { API_ENDPOINTS } from '../api/endpoints'
import type { ApiResponse, InspectionItem as BaseInspectionItem } from '@sstcp/shared'

export interface InspectionItem extends BaseInspectionItem {
  children?: InspectionItem[]
}

export interface InspectionItemCreate {
  item_code: string
  item_name: string
  item_type: string
  level?: number
  parent_id?: number | null
  check_content?: string
  check_standard?: string
  sort_order?: number
}

export interface InspectionItemUpdate {
  item_code?: string
  item_name?: string
  item_type?: string
  level?: number
  parent_id?: number | null
  check_content?: string
  check_standard?: string
  sort_order?: number
}

export interface PaginatedResponse {
  code: number
  message: string
  data: {
    content: InspectionItem[]
    totalElements: number
    totalPages: number
    size: number
    number: number
    first: boolean
    last: boolean
  }
}

export const inspectionItemService = {
  async getTree(): Promise<ApiResponse<InspectionItem[]>> {
    return await request.get(API_ENDPOINTS.INSPECTION_ITEM.TREE)
  },

  async getAll(): Promise<ApiResponse<InspectionItem[]>> {
    return await request.get(API_ENDPOINTS.INSPECTION_ITEM.ALL)
  },

  async getList(params?: {
    page?: number
    size?: number
    keyword?: string
  }): Promise<PaginatedResponse> {
    return await request.get(API_ENDPOINTS.INSPECTION_ITEM.LIST, { params })
  },

  async getById(id: number): Promise<ApiResponse<InspectionItem>> {
    return await request.get(API_ENDPOINTS.INSPECTION_ITEM.DETAIL(id))
  },

  async create(data: InspectionItemCreate): Promise<ApiResponse<InspectionItem>> {
    return await request.post(API_ENDPOINTS.INSPECTION_ITEM.LIST, data)
  },

  async update(id: number, data: InspectionItemUpdate): Promise<ApiResponse<InspectionItem>> {
    return await request.put(API_ENDPOINTS.INSPECTION_ITEM.DETAIL(id), data)
  },

  async delete(id: number): Promise<ApiResponse<null>> {
    return await request.delete(API_ENDPOINTS.INSPECTION_ITEM.DETAIL(id))
  },
}

File: backend-python/app/schemas/inspection_item.py
from datetime import datetime

from pydantic import BaseModel, Field


class InspectionItemBase(BaseModel):
    item_code: str = Field(..., max_length=50, description='事项编码')
    item_name: str = Field(..., max_length=200, description='事项名称')
    item_type: str = Field(..., max_length=50, description='事项类型')
    level: int = Field(1, ge=1, le=3, description='层级: 1-项目类型, 2-系统类型, 3-检查项')
    parent_id: int | None = Field(None, description='父节点ID')
    check_content: str | None = Field(None, description='检查内容')
    check_standard: str | None = Field(None, description='检查标准')
    sort_order: int = Field(0, description='排序')

class InspectionItemCreate(InspectionItemBase):
    pass

class InspectionItemUpdate(BaseModel):
    item_code: str | None = Field(None, max_length=50, description='事项编码')
    item_name: str | None = Field(None, max_length=200, description='事项名称')
    item_type: str | None = Field(None, max_length=50, description='事项类型')
    level: int | None = Field(None, ge=1, le=3, description='层级')
    parent_id: int | None = Field(None, description='父节点ID')
    check_content: str | None = Field(None, description='检查内容')
    check_standard: str | None = Field(None, description='检查标准')
    sort_order: int | None = Field(None, description='排序')

class InspectionItem(InspectionItemBase):
    id: int
    created_at: datetime
    updated_at: datetime
    children: list['InspectionItem'] = []

    class Config:
        from_attributes = True

InspectionItem.model_rebuild()
